File: soluciones/java/Ejercicio13.java
/*
 * Ejercicio 13 - Nivel Avanzado
 * boolean esPalindromo(int[] arr, int tam)
 * Determina si un arreglo es simetrico (se lee igual en ambos sentidos).
 */
import java.util.Scanner;

public class Ejercicio13 {

    // En Java el tipo logico se llama 'boolean' (con valores true / false).
    static boolean esPalindromo(int[] arr, int tam) {
        for (int i = 0; i < tam / 2; i++) {
            if (arr[i] != arr[tam - 1 - i]) {
                return false;
            }
        }
        return true;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Ingrese el tamano del arreglo: ");
        int tam = sc.nextInt();

        int[] arr = new int[tam];
        for (int i = 0; i < tam; i++) {
            System.out.print("Ingrese el elemento " + i + ": ");
            arr[i] = sc.nextInt();
        }

        if (esPalindromo(arr, tam)) {
            System.out.println("El arreglo es simetrico (palindromo).");
        } else {
            System.out.println("El arreglo NO es simetrico.");
        }
        sc.close();
    }
}

File: soluciones/cpp/ejercicio13.cpp
/*
 * Ejercicio 13 - Nivel Avanzado
 * bool esPalindromo(int Arr[], int tam)
 * Determina si un arreglo es simetrico (se lee igual en ambos sentidos).
 */
#include <iostream>
using namespace std;   /* En C++ bool, true y false son parte del lenguaje. */

bool esPalindromo(int arr[], int tam) {
    for (int i = 0; i < tam / 2; i++) {
        if (arr[i] != arr[tam - 1 - i]) {
            return false;
        }
    }
    return true;
}

int main() {
    int tam;
    cout << "Ingrese el tamano del arreglo: ";
    cin >> tam;

    int* arr = new int[tam];
    for (int i = 0; i < tam; i++) {
        cout << "Ingrese el elemento " << i << ": ";
        cin >> arr[i];
    }

    if (esPalindromo(arr, tam)) {
        cout << "El arreglo es simetrico (palindromo)." << endl;
    } else {
        cout << "El arreglo NO es simetrico." << endl;
    }

    delete[] arr;
    return 0;
}
